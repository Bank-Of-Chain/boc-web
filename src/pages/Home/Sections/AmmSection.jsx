import React from "react"
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles"

// === Components === //
import Amms from "../../../components/Amms/index"

import styles from "./ammStyle"

const useStyles = makeStyles(styles)

export default function AmmSection () {
  const classes = useStyles()
  return (
    <div className={classes.section}>
      <h2 className={classes.title}>Integrated with the 3 major chains, 7 Stablecoins, and optimized among 58 AMM and lending pools</h2>
      <Amms />
    </div>
  )
}
