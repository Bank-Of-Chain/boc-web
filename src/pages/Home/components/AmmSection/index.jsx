import React from 'react'
// @material-ui/core components
import { makeStyles } from '@material-ui/core/styles'

// === Components === //
import Tabs from '@/components/CustomTabs/CustomTabs'
import GridItem from '@/components/Grid/GridItem'

// === Constants === //
import AMMS, { PLATFORM_HOME_URL } from '@/constants/amms'
import CHAINS from '@/constants/chains'
import STABLECOINS from '@/constants/tokens'

// === Utils === //
import map from 'lodash/map'
import isEmpty from 'lodash/isEmpty'

// === Styles === //
import styles from './ammStyle'

const useStyles = makeStyles(styles)

export default function AmmSection() {
  const classes = useStyles()
  const [value, setValue] = React.useState(0)

  const handleChange = (event, newValue) => {
    setValue(newValue)
  }

  function imgError(e) {
    const evn = e
    const img = evn.srcElement ? evn.srcElement : evn.target
    img.src = '/default.png'
  }

  return (
    <div className={classes.section}>
      <h2>Major integrations</h2>
      <h4 className={classes.title}>
        Bank of Chain integrates with the major {CHAINS.length > 1 ? 'chains' : 'chain'} of DeFi and the worldâ€™s top {STABLECOINS.length} stablecoins,
        optimizing {AMMS.length} AMM and lending pools.
      </h4>
      <Tabs
        centered
        value={value}
        indicatorColor="primary"
        textColor="primary"
        onChange={handleChange}
        aria-label="disabled tabs example"
        tabs={[
          {
            tabName: 'Stablecoins',
            tabContent: (
              <GridItem xs={12} sm={12} md={12} className={classes.iconContainer}>
                {map(STABLECOINS, c => (
                  <div key={c.id} className={`${classes.item}`} onClick={() => window.open(`https://etherscan.io/address/${c.address}`)}>
                    <img className={classes.img} src={`/images/${c.address}.png`} alt={c.symbol} />
                    <p className={classes.text}>{c.symbol}</p>
                  </div>
                ))}
              </GridItem>
            )
          },
          {
            tabName: 'Major Chain(s)',
            tabContent: (
              <GridItem xs={12} sm={12} md={12} className={classes.iconContainer}>
                {map(CHAINS, c => (
                  <div key={c.id} className={classes.item} onClick={() => window.open(c.url)}>
                    <img className={`${classes.img} ${classes.transparentBg}`} src={`/images/chains/${c.id}.png`} alt={c.name} />
                    <p className={classes.text}>{c.name}</p>
                  </div>
                ))}
              </GridItem>
            )
          },
          {
            tabName: 'Protocols',
            tabContent: (
              <GridItem xs={12} sm={12} md={12} className={classes.iconContainer}>
                {map(
                  map(AMMS, (item, i) => {
                    return {
                      name: item,
                      index: i,
                      url: PLATFORM_HOME_URL[item]
                    }
                  }),
                  a => (
                    <div key={a.name} className={classes.item} onClick={() => !isEmpty(a.url) && window.open(a.url)}>
                      <img className={classes.img} src={`/images/amms/${a.name}.png`} alt={a.name} onError={imgError} />
                      <p className={classes.text}>{a.name}</p>
                    </div>
                  )
                )}
              </GridItem>
            )
          }
        ]}
      />
    </div>
  )
}
