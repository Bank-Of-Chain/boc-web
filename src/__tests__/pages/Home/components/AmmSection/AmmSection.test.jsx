import React from 'react'
import { render } from '@testing-library/react'
import AmmSection from '@/pages/Home/components/AmmSection'

test('AmmSection Component Render', () => {
  const { asFragment } = render(<AmmSection />)
  expect(asFragment()).toMatchSnapshot()
})
