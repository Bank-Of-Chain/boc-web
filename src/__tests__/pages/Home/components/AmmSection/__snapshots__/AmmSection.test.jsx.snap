// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AmmSection Component Render 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-section-1"
  >
    <h2>
      Major integrations
    </h2>
    <h4
      class="makeStyles-title-3"
    >
      Bank of Chain integrates with the major chain of DeFi and the worldâ€™s top 8 stablecoins, optimizing 14 AMM and lending pools.
    </h4>
    <div
      class="MuiTabs-root makeStyles-tabsRoot-13"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          class="MuiTabs-flexContainer MuiTabs-centered"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root makeStyles-tabRootButton-14 MuiTab-textColorInherit Mui-selected makeStyles-tabSelected-15"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper makeStyles-tabWrapper-16"
            >
              Stablecoins
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root makeStyles-tabRootButton-14 MuiTab-textColorInherit"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiTab-wrapper makeStyles-tabWrapper-16"
            >
              Major Chain(s)
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root makeStyles-tabRootButton-14 MuiTab-textColorInherit"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiTab-wrapper makeStyles-tabWrapper-16"
            >
              Protocols
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-17 PrivateTabIndicator-colorSecondary-19 MuiTabs-indicator makeStyles-displayNone-12"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div>
      <div
        class="MuiGrid-root makeStyles-grid-21 makeStyles-iconContainer-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-12"
      >
        <div
          class="makeStyles-item-4"
        >
          <img
            alt="DAI"
            class="makeStyles-img-6"
            src="/images/0x6B175474E89094C44Da98b954EedeAC495271d0F.png"
          />
          <p
            class="makeStyles-text-8"
          >
            DAI
          </p>
        </div>
        <div
          class="makeStyles-item-4"
        >
          <img
            alt="USDC"
            class="makeStyles-img-6"
            src="/images/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48.png"
          />
          <p
            class="makeStyles-text-8"
          >
            USDC
          </p>
        </div>
        <div
          class="makeStyles-item-4"
        >
          <img
            alt="USDT"
            class="makeStyles-img-6"
            src="/images/0xdAC17F958D2ee523a2206206994597C13D831ec7.png"
          />
          <p
            class="makeStyles-text-8"
          >
            USDT
          </p>
        </div>
        <div
          class="makeStyles-item-4"
        >
          <img
            alt="BUSD"
            class="makeStyles-img-6"
            src="/images/0x4Fabb145d64652a948d72533023f6E7A623C7C53.png"
          />
          <p
            class="makeStyles-text-8"
          >
            BUSD
          </p>
        </div>
        <div
          class="makeStyles-item-4"
        >
          <img
            alt="TUSD"
            class="makeStyles-img-6"
            src="/images/0x0000000000085d4780B73119b644AE5ecd22b376.png"
          />
          <p
            class="makeStyles-text-8"
          >
            TUSD
          </p>
        </div>
        <div
          class="makeStyles-item-4"
        >
          <img
            alt="LUSD"
            class="makeStyles-img-6"
            src="/images/0x5f98805A4E8be255a32880FDeC7F6728C6568bA0.png"
          />
          <p
            class="makeStyles-text-8"
          >
            LUSD
          </p>
        </div>
        <div
          class="makeStyles-item-4"
        >
          <img
            alt="GUSD"
            class="makeStyles-img-6"
            src="/images/0x056Fd409E1d7A124BD7017459dFEa2F387b6d5Cd.png"
          />
          <p
            class="makeStyles-text-8"
          >
            GUSD
          </p>
        </div>
        <div
          class="makeStyles-item-4"
        >
          <img
            alt="SUSD"
            class="makeStyles-img-6"
            src="/images/0x57Ab1ec28D129707052df4dF418D58a2D46d5f51.png"
          />
          <p
            class="makeStyles-text-8"
          >
            SUSD
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
